File: sentimental-cash/cash.py
from cs50 import get_float


def main():
    while True:
        dollars = get_float("change: ")
        if dollars >= 0:
            break

    cents = int(dollars * 100)

    quarters = calculate_quarters(cents)
    cents -= quarters * 25

    dimes = calculate_dimes(cents)
    cents -= dimes * 10

    nickels = calculate_nickels(cents)
    cents -= nickels * 5

    pennies = calculate_pennies(cents)
    cents -= pennies * 1

    sum = quarters + dimes + nickels + pennies
    print(f" {sum}")

    return dollars


def calculate_quarters(cents):
    quarters = cents // 25
    return quarters


def calculate_dimes(cents):
    dimes = cents // 10
    return dimes


def calculate_nickels(cents):
    nickels = cents // 5
    return nickels


def calculate_pennies(cents):
    pennies = cents // 1
    return pennies


main()

File: cash/cash.c
#include <cs50.h>
#include <stdio.h>

int calculate_quarters(int cents);
int calculate_dimes(int cents);
int calculate_nickels(int cents);
int calculate_pennies(int cents);

int main(void)
{
    int cents;
    do
    {
        cents = get_int("Change owed: ");
    }
    while (cents < 0);

    int quarters = calculate_quarters(cents);
    cents = cents - (quarters * 25);

    int dimes = calculate_dimes(cents);
    cents = cents - (dimes * 10);

    int nickels = calculate_nickels(cents);
    cents = cents - (nickels * 5);

    int pennies = calculate_pennies(cents);
    cents = cents - (pennies * 1);

    printf("%i\n", quarters + dimes + nickels + pennies);
}

int calculate_quarters(int cents)
{
    int quarters;
    for (quarters = 0; cents >= 25; quarters++)
    {
        cents -= 25;
    }
    return quarters;
}

int calculate_dimes(int cents)
{
    int dimes;
    for (dimes = 0; cents >= 10; dimes++)
    {
        cents -= 10;
    }
    return dimes;
}

int calculate_nickels(int cents)
{
    int nickels;
    for (nickels = 0; cents >= 5; nickels++)
    {
        cents -= 5;
    }
    return nickels;
}

int calculate_pennies(int cents)
{
    int pennies;
    for (pennies = 0; cents >= 1; pennies++)
    {
        cents -= 1;
    }
    return pennies;
}
